inc/driver-models/Pin.h

```diff
@@ -425,6 +425,24 @@ namespace codal
         {
             return DEVICE_NOT_IMPLEMENTED;
         }
+
+        /**
+         * Set pin value iff its current value as input is the opposite.
+         * 
+         * If pin is configured as input and reads as !value, set it to value
+         * and return DEVICE_OK.
+         * Otherwise, do nothing and return DEVICE_BUSY.
+         * Note, that this is overwritten in hardware-specific classes to check the condition immedietly before changing the pin value.
+         */
+        virtual int getAndSetDigitalValue(int value)
```

Comment:
Is this any faster than a get followed by a set?

(I assume the purpose is speed)

Reply:
Oh I see, overridden in target